plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftConnectorTest.java

```diff
@@ -650,6 +650,80 @@ public void testInsertRowConcurrently()
         abort("Test fails with a timeout sometimes and is flaky");
     }
 
+    @Test
+    public void testJoinPushdownWithImplicitCast()
+    {
+        try (TestTable leftTable = new TestTable(
+                getQueryRunner()::execute,
+                "left_table_",
+                "(id int, c_boolean boolean, c_tinyint tinyint, c_smallint smallint, c_integer integer, c_bigint bigint, c_real real, c_double_precision double precision, c_decimal_10_2 decimal(10, 2))",
+                ImmutableList.of("(11, true, 12, 12, 12, 12, 12.34, 12.34, 12.34)", "(12, false, 123, 123, 123, 123, 123.67, 123.67, 123.67)"));
+                TestTable rightTable = new TestTable(
+                        getQueryRunner()::execute,
+                        "right_table_",
+                        "(id int, c_boolean boolean, c_tinyint tinyint, c_smallint smallint, c_integer integer, c_bigint bigint, c_real real, c_double_precision double precision, c_decimal_10_2 decimal(10, 2))",
+                        ImmutableList.of("(21, true, 12, 12, 12, 12, 12.34, 12.34, 12.34)", "(22, true, 234, 234, 234, 234, 234.67, 234.67, 234.67)"))) {
+            Session session = joinPushdownEnabled(getSession());
+            String joinQuery = "SELECT l.id FROM " + leftTable.getName() + " l %s " + rightTable.getName() + " r ON %s";
+
+            assertThat(query(session,joinQuery.formatted("LEFT JOIN", "l.c_tinyint = r.c_bigint")))
+                    .isFullyPushedDown();
+            assertThat(query(session,joinQuery.formatted("RIGHT JOIN", "l.c_tinyint = r.c_bigint")))
+                    .isFullyPushedDown();
+            assertThat(query(session,joinQuery.formatted("INNER JOIN", "l.c_tinyint = r.c_bigint")))
+                    .isFullyPushedDown();
+            assertThat(query(session,joinQuery.formatted("FULL JOIN", "l.c_tinyint = r.c_bigint")))
+                    .isFullyPushedDown();
```

Comment:
full join test fails for me locally. It could be due to https://github.com/trinodb/trino/blob/master/plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java#L432-L435

Reply:
you may want to temporarily add connector test in CI like https://github.com/trinodb/trino/pull/23490/commits/189caa3c735043a9f55c4b81b86ccb0e85217e62